#! /bin/sh /usr/share/dpatch/dpatch-run
## fix-ctime-warnings.dpatch by Phillip Susi <psusi@cfl.rr.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Use local time_t variables rather than casting a u32 * to
## DP: time_t *, which causes warnings and trouble if time_t
## DP: is not 32 bits

@DPATCH@
diff -urNad defrag-0.73pjm1~/e2dump.c defrag-0.73pjm1/e2dump.c
--- defrag-0.73pjm1~/e2dump.c	2006-11-15 14:26:18.000000000 -0500
+++ defrag-0.73pjm1/e2dump.c	2006-11-15 14:27:08.000000000 -0500
@@ -449,6 +449,7 @@
 
 static void dump_super(void)
 {
+  time_t t;
   printf("SUPERBLOCK:\n");
   printf("Inodes count:%lu\n", 
 	 (unsigned long) s.s_inodes_count);      /* Inodes count */
@@ -470,10 +471,12 @@
 	 (unsigned long) s.s_frags_per_group);   /* # Fragments per group */
   printf("Inodes per group:%lu\n",
 	 (unsigned long) s.s_inodes_per_group);  /* # Inodes per group */
+  t = s.s_mtime;
   printf("mount time:%s", 
-	 ctime ((time_t *) &s.s_mtime));    /* Mount time */
+	 ctime (&t));    /* Mount time */
+  t = s.s_wtime;
   printf("write time:%s", 
-	 ctime ((time_t *) &s.s_wtime));    /* Write time */
+	 ctime (&t));    /* Write time */
   printf("magic:0x%X", s.s_magic); 		       /* Magic signature */
   if (s.s_magic == EXT2_SUPER_MAGIC) printf(" (OK)\n");
   else printf(" (???) ");
@@ -488,6 +491,7 @@
      those when on a 64-bit system, though that requires some autoconf magic
      to find the right printf directive. */
    struct ext2_inode n;
+   time_t t;
 
    printf("\nINODE %lu\n", (unsigned long) inode_no);
    load_inode(&n,inode_no);
@@ -503,9 +507,12 @@
    printf("Owner Uid %d ",(uint) n.i_uid);
    printf("Group Id %d\n",(uint) n.i_gid);
    printf("File size %lu\n", (unsigned long) n.i_size);
-   printf("Access time      : %s", ctime ((time_t *) &n.i_atime));
-   printf("Creation time    : %s", ctime ((time_t *) &n.i_ctime));
-   printf("Modification time: %s", ctime ((time_t *) &n.i_mtime));
+   t = n.i_atime;
+   printf("Access time      : %s", ctime (&t));
+   t = n.i_ctime;
+   printf("Creation time    : %s", ctime (&t));
+   t = n.i_mtime;
+   printf("Modification time: %s", ctime (&t));
    if (inode_is_busy(inode_no)) {
       if (n.i_dtime!=0) printf("ERROR: bitmap is 1  ");
    }
@@ -513,7 +520,10 @@
       if (n.i_dtime==0 && n.i_ctime!=0) printf("ERROR: bitmap is 0\n ");
       
    if (n.i_dtime!=0)
-      printf("Deletion time    : %s", ctime ((time_t *) &n.i_dtime));
+     {
+       t = n.i_dtime;
+       printf("Deletion time    : %s", ctime (&t));
+     }
    printf("Links count: %d\n", (uint) n.i_links_count);
                              /* in 512 byte blocks for some unknown reason */
    printf("512-Blocks count: %lu\n", (unsigned long) n.i_blocks); 
